In [1]:
%store -r overall_loc_scores

import pandas as pd
import numpy as np
from scipy.stats import ttest_ind

In [2]:
overall_loc_scores = overall_loc_scores.fillna(0)

## T-test for User Requirement Score

In [3]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','user-requirement-phase1', 'user-requirement-phase2', 'user-requirement-phase3', 'user-requirement-phase4', 'user-requirement-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
user-requirement-phase1     0.0     1.0 -2.647310  0.004998
user-requirement-phase2     0.0     1.0 -2.336790  0.011250
user-requirement-phase4     0.0     1.0 -2.084736  0.020193
user-requirement-phase2     0.0     2.0 -1.818394  0.036638
user-requirement-phase1     0.0     3.0 -2.042718  0.022302
user-requirement-phase2     0.0     3.0 -3.025983  0.001768
user-requirement-phase5     0.0     3.0 -1.836995  0.035004

In [4]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','user-requirement-phase1', 'user-requirement-phase2', 'user-requirement-phase3', 'user-requirement-phase4', 'user-requirement-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

## T-test for Novelty Score

In [5]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','novelty-phase1', 'novelty-phase2', 'novelty-phase3', 'novelty-phase4', 'novelty-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2    tstat    pvalue
novelty-phase3     1.0     3.0  1.89202  0.031161

In [6]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','novelty-phase1', 'novelty-phase2', 'novelty-phase3', 'novelty-phase4', 'novelty-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
novelty-phase1     0.0     1.0 -3.017249  0.001753
novelty-phase2     0.0     1.0 -1.734080  0.043493
novelty-phase3     0.0     1.0 -1.690709  0.047455
novelty-phase1     0.0     2.0 -2.738388  0.003843
novelty-phase1     0.0     3.0 -2.029446  0.022925
novelty-phase5     0.0     3.0 -1.714925  0.045142

## T-test for Infovis Score

In [7]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','infovis-phase1', 'infovis-phase2', 'infovis-phase3', 'infovis-phase4', 'infovis-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

In [8]:
df = overall_loc_scores[['Group','infovis-phase1', 'infovis-phase2', 'infovis-phase3', 'infovis-phase4', 'infovis-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
infovis-phase1     0.0     1.0 -4.002261  0.000073
infovis-phase2     0.0     1.0 -2.402782  0.009356
infovis-phase4     0.0     1.0 -2.027226  0.023007
infovis-phase5     0.0     1.0 -1.777497  0.039672
infovis-phase1     0.0     2.0 -3.098614  0.001342
infovis-phase1     0.0     3.0 -3.825048  0.000130
infovis-phase2     0.0     3.0 -2.245889  0.013783
infovis-phase5     0.0     3.0 -2.361734  0.010328

## T-test for Tech Score

In [9]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','tech-phase1', 'tech-phase2', 'tech-phase3', 'tech-phase4', 'tech-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
tech-phase2     0.0     1.0 -2.373259  0.010044
tech-phase3     0.0     1.0 -2.157914  0.017014
tech-phase4     0.0     1.0 -2.168441  0.016625
tech-phase2     0.0     2.0 -2.136424  0.017860
tech-phase3     0.0     2.0 -1.912652  0.029733
tech-phase5     0.0     2.0 -2.185207  0.015954
tech-phase2     0.0     3.0 -2.945232  0.002141
tech-phase3     0.0     3.0 -2.746689  0.003776
tech-phase4     0.0     3.0 -2.701386  0.004266
tech-phase5     0.0     3.0 -2.824127  0.003057

In [10]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','tech-phase1', 'tech-phase2', 'tech-phase3', 'tech-phase4', 'tech-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

## T-test for Total Score

In [11]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','total-phase1', 'total-phase2', 'total-phase3', 'total-phase4', 'total-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

In [12]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','total-phase1', 'total-phase2', 'total-phase3', 'total-phase4', 'total-phase5']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
total-phase1     0.0     1.0 -3.425757  0.000502
total-phase2     0.0     1.0 -2.591738  0.005739
total-phase3     0.0     1.0 -2.014142  0.023722
total-phase4     0.0     1.0 -2.279447  0.012674
total-phase1     0.0     2.0 -2.406625  0.009217
total-phase2     0.0     2.0 -2.083155  0.020258
total-phase1     0.0     3.0 -2.882155  0.002543
total-phase2     0.0     3.0 -2.444318  0.008380
total-phase4     0.0     3.0 -1.724428  0.044275
total-phase5     0.0     3.0 -2.481327  0.007612

## T-test for improve-novel

In [13]:
df = overall_loc_scores[['Group','improve-novel-12', 'improve-novel-23', 'improve-novel-34', 'improve-novel-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

In [14]:
df = overall_loc_scores[['Group','improve-novel-12', 'improve-novel-23', 'improve-novel-34', 'improve-novel-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-novel-45     1.0     3.0 -1.897917  0.030894

## T-test for improve-user-requirement

In [15]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-ur-12', 'improve-ur-23', 'improve-ur-34', 'improve-ur-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-ur-45     1.0     3.0 -2.003729  0.024243
improve-ur-45     2.0     3.0 -1.717945  0.044820

In [16]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-ur-12', 'improve-ur-23', 'improve-ur-34', 'improve-ur-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy = 'omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2    tstat    pvalue
improve-ur-23     0.0     3.0  2.43615  0.008747

## T-tests for improve-vis score

In [17]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-vis-12', 'improve-vis-23', 'improve-vis-34', 'improve-vis-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-vis-12     0.0     1.0  2.266028  0.013099
improve-vis-12     0.0     2.0  2.290139  0.012490
improve-vis-12     0.0     3.0  2.006777  0.024149

In [18]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-vis-12', 'improve-vis-23', 'improve-vis-34', 'improve-vis-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-vis-45     0.0     3.0 -2.181642  0.016102
improve-vis-45     1.0     3.0 -2.514000  0.007001

## T-test for improve-tech 

In [19]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-tech-12', 'improve-tech-23', 'improve-tech-34', 'improve-tech-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

In [20]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-tech-12', 'improve-tech-23', 'improve-tech-34', 'improve-tech-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-tech-45     0.0     2.0 -1.847378  0.034915
improve-tech-45     1.0     2.0 -2.581302  0.005874
improve-tech-45     1.0     3.0 -2.023538  0.023387

## T-test for improve-total

In [21]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-total-12', 'improve-total-23', 'improve-total-34', 'improve-total-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'less', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

group1  group2     tstat    pvalue
improve-total-45     0.0     3.0 -1.728091  0.043945
improve-total-45     1.0     3.0 -3.168493  0.001128

In [22]:
from itertools import combinations
from scipy import stats as st

df = overall_loc_scores[['Group','improve-total-12', 'improve-total-23', 'improve-total-34', 'improve-total-45']]

def all_pairwise(df, compare_col = 'Group'):
    decade_pairs = [(i,j) for i, j in combinations(df[compare_col].unique().tolist(), 2)]
    # or add a list of colnames to function signature
    cols = list(df.columns)
    cols.remove(compare_col)
    list_of_dfs = []
    for pair in decade_pairs:
        for col in cols:
            c1 = df[df[compare_col] == pair[0]][col]
            c2 = df[df[compare_col] == pair[1]][col]
            results = st.ttest_ind(c1, c2, nan_policy='omit', alternative = 'greater', equal_var = False)
            tmp = pd.DataFrame({'group1': pair[0],
                                'group2': pair[1],
                                'tstat': results.statistic,
                                'pvalue': results.pvalue}, index = [col])
            list_of_dfs.append(tmp)
    df_stats = pd.concat(list_of_dfs)
    return df_stats

df_stats = all_pairwise(df)
df_stats[df_stats['pvalue'] < 0.05]

Empty DataFrame
Columns: [group1, group2, tstat, pvalue]
Index: []

## Comments on Improvement Score

* Improvement Score Novelty

> * Significantly less improvement in novelty in phase 4 - 5 ( 0.030894 ) in performance transparency compared to full transparency. 

* Improvement Score User Requirement 

> * Significantly less improvement in user requirement in phase 4 - 5 ( 0.024243 ) in performance transparency compared to full transparency. 

> * Singificantly less improvement score in user requirement phase 4 - 5 ( 0.044820 ) in solution transparency compared to full transparency 

> * Significantly **higher** improvement score in user requirement phase 2 - 3 ( 0.008747 ) in no transparency compared to full transparency

* Improvement Score Infovis 

> * Significantly less improvement in info vis in phase 4 - 5 ( 0.016102 ) in no transparency compared to full transparency. 

> * Significantly less improvement in info vis in phase 4 - 5 ( 0.007001 ) in performance transparency cimpared to full transparency. 

> * Significantly **higher** improvement in info vis in phase 1 - 2 ( 0.013099 ) in no transparency compared to performance transparency. 

> * Significantly **higher** improvement in info vis in phase 1 - 2 ( 0.012490 ) in no transparency compared to solution transparency. 

> * Significantly **higher** improvement in info vis in phase 1 - 2 ( 0.024149 ) in no transparency compared to full transparency. 

* Improvement Score Technology

> * Significantly less improvement in technology in phase 4 - 5 ( 0.034915 ) in no transparency compared to solution transparency. 

> * Significantly less improvement in technology in phase 4 - 5 ( 0.005874 ) in no transparency compared to performance transparency compared to solution transparency. 

> * Significantly less improvement in technology in phase 4 - 5 ( 0.023387 ) in performance transparency compared to full transparency. 

* Improvement Score Total

> * Significantly less improvement in total sore in phase 4 - 5 ( 0.043945 ) in no transparency compared to full transparency. 

> * Significantly less improvement in total score in phase 4 - 5 ( 0.001128 ) in performance transparency compared to full transparency. 

## Comments on Absolute Score

> Tehcnology Score ( No Transprency significantly less than all others in the following phases )

* tech-phase2	0.0	1.0	-2.373259	0.010044
* tech-phase3	0.0	1.0	-2.157914	0.017014
* tech-phase4	0.0	1.0	-2.168441	0.016625
* tech-phase2	0.0	2.0	-2.136424	0.017860
* tech-phase3	0.0	2.0	-1.912652	0.029733
* tech-phase5	0.0	2.0	-2.185207	0.015954
* tech-phase2	0.0	3.0	-2.945232	0.002141
* tech-phase3	0.0	3.0	-2.746689	0.003776
* tech-phase4	0.0	3.0	-2.701386	0.004266
* tech-phase5	0.0	3.0	-2.824127	0.003057

> Info vis score ( No Transparency significantly less than all others in the following phase ) 

* infovis-phase1	0.0	1.0	-4.002261	0.000073
* infovis-phase2	0.0	1.0	-2.402782	0.009356
* infovis-phase4	0.0	1.0	-2.027226	0.023007
* infovis-phase5	0.0	1.0	-1.777497	0.039672
* infovis-phase1	0.0	2.0	-3.098614	0.001342
* infovis-phase1	0.0	3.0	-3.825048	0.000130
* infovis-phase2	0.0	3.0	-2.245889	0.013783
* infovis-phase5	0.0	3.0	-2.361734	0.010328

> Novelty score ( No Transparency significantly less than all others in the following phase ) 

* novelty-phase1	0.0	1.0	-3.017249	0.001753
* novelty-phase2	0.0	1.0	-1.734080	0.043493
* novelty-phase3	0.0	1.0	-1.690709	0.047455
* novelty-phase1	0.0	2.0	-2.738388	0.003843
* novelty-phase1	0.0	3.0	-2.029446	0.022925
* novelty-phase5	0.0	3.0	-1.714925	0.045142

> Novelty score ( Performance Transparency significantly higher than full transparency in phase 3 ) 

* novelty-phase3	1.0	3.0	1.89202	0.031161

> User Requirement Score ( No Transparency significantly less than all others in the following phase ) 

* user-requirement-phase1	0.0	1.0	-2.647310	0.004998
* user-requirement-phase2	0.0	1.0	-2.336790	0.011250
* user-requirement-phase4	0.0	1.0	-2.084736	0.020193
* user-requirement-phase2	0.0	2.0	-1.818394	0.036638
* user-requirement-phase1	0.0	3.0	-2.042718	0.022302
* user-requirement-phase2	0.0	3.0	-3.025983	0.001768
* user-requirement-phase5	0.0	3.0	-1.836995	0.035004

> Total Score ( No Transparency significantly less than all others in the following phase ) 

* total-phase1	0.0	1.0	-3.425757	0.000502
* total-phase2	0.0	1.0	-2.591738	0.005739
* total-phase3	0.0	1.0	-2.014142	0.023722
* total-phase4	0.0	1.0	-2.279447	0.012674
* total-phase1	0.0	2.0	-2.406625	0.009217
* total-phase2	0.0	2.0	-2.083155	0.020258
* total-phase1	0.0	3.0	-2.882155	0.002543
* total-phase2	0.0	3.0	-2.444318	0.008380
* total-phase4	0.0	3.0	-1.724428	0.044275
* total-phase5	0.0	3.0	-2.481327	0.007612